src/Compatibility/Microsoft.DotNet.ApiSymbolExtensions/Filtering/SymbolFilterFactory.cs

```diff
@@ -0,0 +1,76 @@
+                                                                 bool includeExplicitInterfaceImplementationSymbols = true)
+    {
+        DocIdSymbolFilter? docIdSymbolFilter =
+            apiExclusionFilePaths?.Count() > 0 ?
+            DocIdSymbolFilter.CreateFromFiles(apiExclusionFilePaths) : null;
+
+        return GetCompositeSymbolFilter(docIdSymbolFilter, respectInternals, includeEffectivelyPrivateSymbols, includeExplicitInterfaceImplementationSymbols, withImplicitSymbolFilter: true);
+    }
+
+    /// <summary>
+    /// Creates a composite filter to exclude APIs using the DocIDs provided in the specifed list.
+    /// </summary>
+    /// <param name="apiExclusionList">A collection of exclusion list.</param>
+    /// <param name="respectInternals">Whether to include internal symbols or not.</param>
+    /// <param name="includeEffectivelyPrivateSymbols">Whether to include effectively private symbols or not.</param>
+    /// <param name="includeExplicitInterfaceImplementationSymbols">Whether to include explicit interface implementation symbols or not.</param>
+    /// <returns>An instance of the symbol filter.</returns>
+    public static ISymbolFilter GetFilterFromList(string[]? apiExclusionList,
+                                                                bool respectInternals = false,
+                                                                bool includeEffectivelyPrivateSymbols = true,
+                                                                bool includeExplicitInterfaceImplementationSymbols = true)
+    {
+        DocIdSymbolFilter? docIdSymbolFilter =
+            apiExclusionList?.Count() > 0 ?
+            DocIdSymbolFilter.CreateFromLists(apiExclusionList) : null;
+
+        return GetCompositeSymbolFilter(docIdSymbolFilter, respectInternals, includeEffectivelyPrivateSymbols, includeExplicitInterfaceImplementationSymbols, withImplicitSymbolFilter: true);
+    }
+
+    private static ISymbolFilter GetCompositeSymbolFilter(DocIdSymbolFilter? customFilter,
+                                                                  bool respectInternals,
+                                                                  bool includeEffectivelyPrivateSymbols,
+                                                                  bool includeExplicitInterfaceImplementationSymbols,
+                                                                  bool withImplicitSymbolFilter)
+    {
+        AccessibilitySymbolFilter accessibilitySymbolFilter = new(
+                respectInternals,
+                includeEffectivelyPrivateSymbols,
+                includeExplicitInterfaceImplementationSymbols);
+
+        CompositeSymbolFilter filter = new();
+
+        if (customFilter != null)
+        {
+            filter.Add(customFilter);
+        }
+        if (withImplicitSymbolFilter)
+        {
+            filter.Add(new ImplicitSymbolFilter());
```

Comment:
Please add unit tests for the `SymbolFilterFactory` types. Make sure that those are real unit tests, not integration tests by using mocks so that only the type in question gets tested. When writing those tests, you might notice that the current implementation won't allow for that.

Reply:
Working on these.

Reply:
Added a couple of tests that verify the implemented `Include` method depending if the symbol needs to be excluded or included.